src/main/java/technology/dice/dicefairlink/discovery/members/awsapi/AwsApiReplicasFinder.java

```diff
@@ -0,0 +1,111 @@
+import software.amazon.awssdk.services.rds.paginators.DescribeDBInstancesIterable;
+import technology.dice.dicefairlink.config.FairlinkConfiguration;
+import technology.dice.dicefairlink.discovery.members.ClusterInfo;
+import technology.dice.dicefairlink.discovery.members.MemberFinderMethod;
+import technology.dice.dicefairlink.driver.FairlinkConnectionString;
+
+import java.util.HashSet;
+import java.util.Optional;
+import java.util.Set;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import java.util.stream.Collectors;
+
+public class AwsApiReplicasFinder implements MemberFinderMethod {
+  private static final Logger LOGGER = Logger.getLogger(AwsApiReplicasFinder.class.getName());
+  private static final String ACTIVE_STATUS = "available";
+  private static final Set<String> EMPTY_SET = new HashSet<>(0);
+  public static final String DB_CLUSTER_ID_FILTER = "db-cluster-id";
+  private final String clusterId;
+  private final RdsClient client;
+
+  public AwsApiReplicasFinder(
+      FairlinkConfiguration fairlinkConfiguration,
+      FairlinkConnectionString fairlinkConnectionString) {
+
+    this.clusterId = fairlinkConnectionString.getHost();
+    LOGGER.log(Level.INFO, "Cluster ID: {0}", fairlinkConnectionString.getHost());
+    LOGGER.log(Level.INFO, "AWS Region: {0}", fairlinkConfiguration.getAuroraClusterRegion());
+    this.client =
+        RdsClient.builder()
+            .region(fairlinkConfiguration.getAuroraClusterRegion())
+            .credentialsProvider(fairlinkConfiguration.getAwsCredentialsProvider())
+            .build();
+  }
+
+  private DBCluster describeCluster(String clusterId) {
+    final DescribeDbClustersResponse describeDbClustersResponse =
+        client.describeDBClusters(
+            DescribeDbClustersRequest.builder().dbClusterIdentifier(clusterId).build());
+    return describeDbClustersResponse.dbClusters().stream()
+        .findFirst()
+        .orElseThrow(
+            () ->
+                new RuntimeException(
+                    String.format(
+                        "Could not find exactly one cluster with cluster id [%s]", clusterId)));
+  }
+
+  private Set<String> replicaMembersOf(DBCluster cluster) {
+    try {
+      DescribeDbInstancesRequest request =
+          DescribeDbInstancesRequest.builder()
+              .filters(
+                  Filter.builder()
+                      .name(DB_CLUSTER_ID_FILTER)
+                      .values(cluster.dbClusterIdentifier())
+                      .build())
+              .build();
+      final Optional<DBClusterMember> writer =
+          cluster.dbClusterMembers().stream().filter(member -> member.isClusterWriter()).findAny();
+      final DescribeDBInstancesIterable describeDbInstancesResponses =
+          client.describeDBInstancesPaginator(request);
+      final Set<String> replicaIds =
+          describeDbInstancesResponses.stream()
+              .flatMap(
+                  dbInstances ->
```

Comment:
`response ->` ?

Reply:
I don't get it

Reply:
Andrey means that a collection named `describeDbInstancesResponses` would reasnably be expected to contain responses. So either rename that to `...Instances` or this one to `response`